### PR TITLE
cam6_4_048: Timestamp in rpointer name

### LINKED ISSUE #1146: Add timestamp to rpointer file name
### What is the feature/what would you like to discuss?

Adding a timestamp to the rpointer file name will
provide greater flexibility in restarting the model 
removing the need to hand edit rpointer files.

### Is there anyone in particular you want to be part of this conversation?

_No response_

### Will this change (regression test) answers?

No

### Will you be implementing this enhancement yourself?

Yes

### PR DESCRIPTION
**Pull request by jedwards4b**
Add a timestamp in the rpointer name.
This is a backward compatible change in that rpointer.cam will still work
but new files will include a timestamp.
Fixes #1146

### CODE REVIEW

**Review by brian-eaton — CHANGES_REQUESTED** [review]
minor comments and a question

File: src/control/cam_initfiles.F90 [hunk 1]
```diff
@@ -112,18 +112,25 @@ subroutine cam_initfiles_readnl(nlfile)
...
    ! Set name of primary restart file
    if (restart_run) then
       ! Read name of restart file from pointer file
       if (masterproc) then
+         rest_pfile = interpret_filename_spec("rpointer.cam"//trim(inst_suffix)//".%y-%m-%d-%s", prev=.true.)
+         inquire(file=trim(rest_pfile),exist=found)
+         if(.not. found) then
+            write(iulog, "Warning : rpointer file "//trim(rest_pfile)//" not found.")
+            rest_pfile = "rpointer.cam"//trim(inst_suffix)
+            inquire(file=trim(rest_pfile),exist=found)
```
[the remark this answers is not in the document]
Comment (jedwards4b) [reply src/control/cam_initfiles.F90]: done

File: src/cpl/nuopc/atm_comp_nuopc.F90 [hunk 2]
```diff
@@ -1967,15 +1967,18 @@ subroutine cam_write_clockrest( clock, yr_spec, mon_spec, day_spec, sec_spec, rc
...
+
     call shr_cal_ymd2date(yy,mm,dd,curr_ymd)
 
     ! Open clock info restart dataset
     restart_file = interpret_filename_spec( '%c.cpl.r.%y-%m-%d-%s.nc', &
          yr_spec=yr_spec, mon_spec=mon_spec, day_spec=day_spec, sec_spec= sec_spec )
 
     if (masterproc) then
+       restart_pfile = interpret_filename_spec('rpointer.cpl.%y-%m-%d-%s',&
+            yr_spec=yr_spec, mon_spec=mon_spec, day_spec=day_spec, sec_spec= sec_spec )
```
[the remark this answers is not in the document]
Comment (jedwards4b) [reply src/cpl/nuopc/atm_comp_nuopc.F90:1980]: Thanks for pointing out this bug, it should replace 'rpointer.cpl' below

**Review by brian-eaton — APPROVED** [review]
looks good now

File: src/control/runtime_opts.F90 [hunk 3]
```diff
@@ -109,7 +109,6 @@ subroutine read_namelist(nlfilename, single_column, scmlat, scmlon)
    logical,          intent(in) :: single_column
    real(r8),         intent(in) :: scmlat
    real(r8),         intent(in) :: scmlon
-
```
[the remark this answers is not in the document]
Comment (jedwards4b) [reply src/control/runtime_opts.F90]: done

**Comment by jedwards4b** [comment]
I'm not sure what happened to the history here.  I rebased to cam_development tag: cam6_4_043 and pushed back.  Now this PR shows my changes applied twice, but git log in my sandbox does not show the original commits.

**Comment by ekluzek** [comment]
This is dependent on the CTSM tag, and updates to: cime, share, cmeps and cdeps.

@peverwhee on Izumi you'll need to update cdeps to cdeps1.0.61, otherwise you'll find problems with the CDEPS tests.

**Pull request merged**